### PR TITLE
[GCS] Remove old lightweight resource usage report code path

### LINKED ISSUE #12591: [GCS] Remove old lightweight heartbeat code path (after new scheduler is enabled).
### Describe your feature request

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?

As new scheduler has been enabled by default, we can remove old code path of light resource usage reporting.
(NOTE: this PR includes lots of renaming in cluster_resource_scheduler_test.cc, which can be ignored while reviewing)

## Related issue number
closes https://github.com/ray-project/ray/issues/12591

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
LGTM if tests pass

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -1030,7 +992,7 @@ void NodeManager::ResourceUsageAdded(const NodeID &node_id,
   }
 }
 
-void NodeManager::ResourceUsageBatchAdded(
+void NodeManager::ResourceUsageBatchReceived(
```
Comment (ffbin) [inline src/ray/raylet/node_manager.cc:995]: How about rename it to `UpdateBatchResourceUsage`?

File: src/ray/raylet/scheduling/cluster_resource_scheduler.cc [hunk 2]
```diff
@@ -912,78 +911,49 @@ void ClusterResourceScheduler::FillResourceUsage(
...
-      // Note: available may be negative, but only report positive to GCS.
-      if (capacity.available > 0) {
-        (*resources_data->mutable_resources_available())[label] =
-            capacity.available.Double();
-      }
-      if (capacity.total != 0) {
-        (*resources_data->mutable_resources_total())[label] = capacity.total.Double();
-      }
+  }
+  for (auto it = resources.custom_resources.begin();
```
Comment (ffbin) [inline src/ray/raylet/scheduling/cluster_resource_scheduler.cc:938]:
```suggestion
  for (const auto &it : resources.custom_resources)
```

File: src/ray/raylet/node_manager.cc [hunk 3]
```diff
@@ -447,76 +445,50 @@ void NodeManager::ReportResourceUsage() {
...
-          (*resources_data->mutable_resource_load())[resource_pair.first] =
-              resource_pair.second;
-        }
-        last_heartbeat_resources->SetLoadResources(
-            ResourceSet(local_resources.GetLoadResources()));
-      }
-    } else {
-      // If light resource usage report disabled, we send whole resources information
-      // every time.
+    // We only set filed that represent resources changed.
```
Comment (ffbin) [inline src/ray/raylet/node_manager.cc:456]:
```suggestion
    // We only set fileds that represent resources changed.
```

File: src/ray/raylet/node_manager.cc (hunk 1 above)
Comment (WangTaoTheTonic) [reply src/ray/raylet/node_manager.cc:995]: This is a callback to handle notification of batched resource usage, so it's better to used `Received`

File: src/ray/raylet/scheduling/cluster_resource_scheduler.cc [hunk 4]
```diff
@@ -912,78 +911,48 @@ void ClusterResourceScheduler::FillResourceUsage(
...
-      }
+  }
+  for (const auto &it : resources.custom_resources) {
+    uint64_t custom_id = it.first;
+    const auto &capacity = it.second;
+    const auto &last_capacity = last_report_resources_->custom_resources[custom_id];
+    const auto &label = string_to_int_map_.Get(custom_id);
+    // Note: available may be negative, but only report positive to GCS.
+    if (capacity.available != last_capacity.available && capacity.available > 0) {
+      resources_data->set_resources_available_changed(true);
```
Comment (ffbin) [inline src/ray/raylet/scheduling/cluster_resource_scheduler.cc:945]: This will be called repeatedly？

Comment (WangTaoTheTonic) [reply src/ray/raylet/scheduling/cluster_resource_scheduler.cc:945]: This is added by @clay4444. Didn't confirm but I guess it's the easiest way to set flag.

**Review by ffbin — APPROVED** [review]

**Review by raulchen — APPROVED** [review]

**Pull request merged**